src/wordcount.js

```diff
@@ -0,0 +1,334 @@
+ *
+ * 		<paragraph>foo</paragraph>
+ * 		<paragraph>bar</paragraph>
+ * 		// Words: 2, Characters: 7
+ *
+ * 		<paragraph><$text bold="true">foo</$text>bar</paragraph>
+ * 		// Words: 1, Characters: 6
+ *
+ * 		<paragraph>*&^%)</paragraph>
+ * 		// Words: 0, Characters: 5
+ *
+ * 		<paragraph>foo(bar)</paragraph>
+ * 		//Words: 2, Characters: 8
+ *
+ * 		<paragraph>12345</paragraph>
+ * 		// Words: 1, Characters: 5
+ *
+ * @extends module:core/plugin~Plugin
+ */
+export default class WordCount extends Plugin {
+	/**
+	 * @inheritDoc
+	 */
+	constructor( editor ) {
+		super( editor );
+
+		/**
+		 * The number of characters in the editor.
+		 *
+		 * @observable
+		 * @readonly
+		 * @member {Number} module:wordcount/wordcount~WordCount#characters
+		 */
+		this.set( 'characters', 0 );
+
+		/**
+		 * The number of words in the editor.
+		 *
+		 * @observable
+		 * @readonly
+		 * @member {Number} module:wordcount/wordcount~WordCount#words
+		 */
+		this.set( 'words', 0 );
+
+		/**
+		 * Label used to display words value in {@link #getWordCountContainer output contianer}
+		 *
+		 * @observable
+		 * @private
+		 * @readonly
+		 * @member {String} module:wordcount/wordcount~WordCount#_wordsLabel
+		 */
+		this.set( '_wordsLabel' );
+
+		/**
+		 * Label used to display characters value in {@link #getWordCountContainer output contianer}
+		 *
+		 * @observable
+		 * @private
+		 * @readonly
+		 * @member {String} module:wordcount/wordcount~WordCount#_charactersLabel
+		 */
+		this.set( '_charactersLabel' );
+
+		/**
+		 * The configuration of this plugins.
+		 *
+		 * @private
+		 * @type {Object}
+		 */
+		this._config = editor.config.get( 'wordCount' ) || {};
+
+		/**
+		 * A reference to a {@link module:ui/view~View view object} which contains self-updating HTML container.
+		 *
+		 * @private
+		 * @readonly
+		 * @type {module:ui/view~View}
+		 */
+		this._outputView;
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	static get pluginName() {
+		return 'WordCount';
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	init() {
+		const editor = this.editor;
+
+		editor.model.document.on( 'change:data', throttle( this._calculateWordsAndCharacters.bind( this ), 250 ) );
+
+		if ( typeof this._config.onUpdate == 'function' ) {
+			this.on( 'update', ( evt, data ) => {
+				this._config.onUpdate( data );
+			} );
+		}
+
+		if ( isElement( this._config.container ) ) {
+			this._config.container.appendChild( this.getWordCountContainer() );
+		}
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	destroy() {
+		this._outputView.element.remove();
+		this._outputView.destroy();
+
+		super.destroy();
+	}
+
+	/**
+	 * Method creates self-updated HTML element. Repeated executions return the same element.
+	 * Returned element has followed HTML structure:
+	 *
+	 * 		<div class="ck ck-word-count">
+	 * 			<div class="ck-word-count__words">Words: 4</div>
+	 * 			<div class="ck-word-count__characters">Characters: 28</div>
+	 * 		</div>
+	 *
+	 * @returns {HTMLElement}
+	 */
+	getWordCountContainer() {
```

Comment:
Couldn't this be a getter? It was proposed as a property in the spec: https://github.com/ckeditor/ckeditor5-word-count/issues/1.